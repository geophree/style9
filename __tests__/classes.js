/* eslint-env jest */
const compile = require('./compile.js');

it('string literals', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue',
    opacity: 1
  },
  red: {
    color: 'red'
  }
});
styles('default', 'red');
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('moves test', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue'
  }
});
styles(foo() && 'default');
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('and', async () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue',
    opacity: 1
  },
  red: {
    color: 'red'
  }
});
styles('default', foo && 'red');
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('ternary', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue',
    opacity: 1
  },
  red: {
    color: 'red'
  }
});
styles(foo ? 'red' : 'default');
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('object', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue',
    opacity: 1
  },
  red: {
    color: 'red'
  }
});
styles({
  default: foo,
  red: bar
});
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('mixed', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue',
    opacity: 1
  },
  red: {
    color: 'red'
  }
});
styles({
  default: foo
}, 'red');
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('property access', () => {
  const input = `
import style9 from 'style9';
const styles1 = style9.create({
  default: {
    color: 'blue'
  }
});
const styles2 = style9.create({
  red: {
    color: 'red'
  }
});
style9(styles1.default, styles2.red)
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('hoists function call', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    color: 'blue'
  }
});
styles({
  default: foo()
})
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('supports destructuring assignment', () => {
  const input = `
import style9 from 'style9';
const { blue } = style9.create({
  blue: {
    color: 'blue'
  }
});
console.log(blue)
  `;
  const { code } = compile(input);
  expect(code).toMatchSnapshot();
});

it('supports member expression access', () => {
  const input = `
import style9 from 'style9';
const blue = style9.create({
  blue: {
    color: 'blue'
  },
  red: {
    color: 'red'
  }
}).blue;
console.log(blue)
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports static bracket access', () => {
  const input = `
import style9 from 'style9';
const blue = style9.create({
  blue: {
    color: 'blue'
  },
  red: {
    color: 'red'
  }
})['blue'];
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports dynamic bracket access', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  blue: {
    color: 'blue'
  },
  red: {
    color: 'red'
  }
})[blue];
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports spread assignment', () => {
  const input = `
import style9 from 'style9';
const { ...styles } = style9.create({
  blue: {
    color: 'blue'
  }
});
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports spread use', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  blue: {
    color: 'blue'
  }
});
console.log({ ...styles });
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});
