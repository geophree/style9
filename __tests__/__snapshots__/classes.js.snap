// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoists function call 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\"
  }
};

const _foo = foo();

_foo ? \\"c1r9f2e5 \\" : \\"\\";"
`;

exports[`mixed 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\",
    opacity: \\"cu2kwdz\\"
  },
  red: {
    color: \\"cyyg6ey\\"
  }
};
(foo ? \\"cu2kwdz \\" : \\"\\") + \\"cyyg6ey \\";"
`;

exports[`moves test 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\"
  }
};

const _foo = foo();

_foo ? \\"c1r9f2e5 \\" : \\"\\";"
`;

exports[`object 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\",
    opacity: \\"cu2kwdz\\"
  },
  red: {
    color: \\"cyyg6ey\\"
  }
};
(bar ? \\"cyyg6ey \\" : foo ? \\"c1r9f2e5 \\" : \\"\\") + (foo ? \\"cu2kwdz \\" : \\"\\");"
`;

exports[`property access 1`] = `
"import style9 from 'style9';
const styles1 = {
  default: {
    color: \\"c1r9f2e5\\"
  }
};
const styles2 = {
  red: {
    color: \\"cyyg6ey\\"
  }
};
style9(styles1.default, styles2.red);"
`;

exports[`string literals 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\",
    opacity: \\"cu2kwdz\\"
  },
  red: {
    color: \\"cyyg6ey\\"
  }
};
\\"cyyg6ey cu2kwdz \\";"
`;

exports[`supports destructuring assignment 1`] = `
"import style9 from 'style9';
const {
  blue
} = {
  blue: {
    color: \\"c1r9f2e5\\"
  }
};
console.log(blue);"
`;

exports[`supports dynamic bracket access 1`] = `
"import style9 from 'style9';
const styles = {
  blue: {
    color: \\"c1r9f2e5\\"
  },
  red: {
    color: \\"cyyg6ey\\"
  }
}[blue];"
`;

exports[`supports dynamic bracket access 2`] = `".c1r9f2e5{color:blue}.cyyg6ey{color:red}"`;

exports[`supports member expression access 1`] = `
"import style9 from 'style9';
const blue = {
  blue: {
    color: \\"c1r9f2e5\\"
  }
}.blue;
console.log(blue);"
`;

exports[`supports member expression access 2`] = `".c1r9f2e5{color:blue}"`;

exports[`supports spread assignment 1`] = `
"import style9 from 'style9';
const { ...styles
} = {
  blue: {
    color: \\"c1r9f2e5\\"
  }
};"
`;

exports[`supports spread assignment 2`] = `".c1r9f2e5{color:blue}"`;

exports[`supports spread use 1`] = `
"import style9 from 'style9';
const styles = {
  blue: {
    color: \\"c1r9f2e5\\"
  }
};
console.log({ ...styles
});"
`;

exports[`supports spread use 2`] = `".c1r9f2e5{color:blue}"`;

exports[`supports static bracket access 1`] = `
"import style9 from 'style9';
const blue = {
  blue: {
    color: \\"c1r9f2e5\\"
  }
}['blue'];"
`;

exports[`supports static bracket access 2`] = `".c1r9f2e5{color:blue}"`;

exports[`ternary 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    color: \\"c1r9f2e5\\",
    opacity: \\"cu2kwdz\\"
  },
  red: {
    color: \\"cyyg6ey\\"
  }
};
(foo ? \\"c1r9f2e5 \\" : \\"cyyg6ey \\") + (foo ? \\"cu2kwdz \\" : \\"\\");"
`;
